security/pkg/nodeagent/cache/secretcache.go

```diff
@@ -235,6 +260,26 @@ func (sc *SecretCache) DeleteSecret(connectionID, resourceName string) {
 	sc.secrets.Delete(key)
 }
 
+// Close shuts down the secret cache.
+func (sc *SecretCache) Close() {
+	sc.closing <- true
+}
+
+func (sc *SecretCache) keyCertRotationJob() {
```

Comment:
Same here, this is not a new function, suggest to move it to the original place for better code review purpose.